src/FS.Embedded/Manifest/ManifestDirectoryInfo.cs

```diff
@@ -0,0 +1,75 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.IO;
+using System.Reflection;
+
+namespace Microsoft.Extensions.FileProviders.Embedded.Manifest
+{
+    internal class ManifestDirectoryInfo : IFileInfo
+    {
+        private DateTimeOffset? _lastModified = null;
+
+        public ManifestDirectoryInfo(Assembly assembly, ManifestDirectory directory)
+        {
+            if (assembly == null)
+            {
+                throw new ArgumentNullException(nameof(assembly));
+            }
+
+            if (directory == null)
+            {
+                throw new ArgumentNullException(nameof(directory));
+            }
+
+            Assembly = assembly;
+            Directory = directory;
+        }
+
+        public bool Exists => true;
+
+        public long Length => -1;
+
+        public string PhysicalPath => null;
+
+        public string Name => Directory.Name;
+
+        public DateTimeOffset LastModified
+        {
+            get
+            {
+                if (_lastModified.HasValue)
+                {
+                    return _lastModified.Value;
+                }
+                _lastModified = DateTimeOffset.UtcNow;
+
+                if (!string.IsNullOrEmpty(Assembly.Location))
+                {
+                    try
+                    {
+                        _lastModified = File.GetLastWriteTimeUtc(Assembly.Location);
```

Comment:
Can we read these values once and pass it through (they're not meant to change once the App loads)?

Reply:
I'm still not happy with this behavior of reading the assembly. I'm giving it some thought